asapdiscovery-simulation/asapdiscovery/simulation/utils.py

```diff
@@ -20,3 +29,148 @@ def create_protein_only_system(input_pdb_path: str, ff_params: ForceFieldParams)
         hydrogenMass=ff_params.hydrogenMass,
     )
     return system
+
+
+class AlchemiscaleHelper:
+    """
+    A convince class to handle alchemiscale submissions restarts and results gathering.
+    """
+
+    def __init__(self, api_url: str = "https://api.alchemiscale.org"):
+        """
+        Create the client which will be used for the rest of the queries
+        """
+        from alchemiscale import AlchemiscaleClient
+
+        # load the settings from the environment
+        settings = AlchemiscaleSettings()
+        # connect to the client
+        self._client = AlchemiscaleClient(
+            api_url=api_url,
+            identifier=settings.ALCHEMISCALE_ID,
+            key=settings.ALCHEMISCALE_KEY,
+        )
+
+    def create_network(
+        self, planned_network: FreeEnergyCalculationNetwork, scope: Scope
+    ) -> FreeEnergyCalculationNetwork:
+        """
+        Create the given network on the alchemiscale instance no tasks are actioned yet.
+
+        Args:
+            planned_network: The network to submit to alchemiscale.
+            scope: The alchemiscale scope the calculation to be submitted to.
+
+        Returns
+            A copy of the network submit with the results object updated.
+        """
+        # store a copy of the input data so we can add the results holder
+        network_data = planned_network.dict()
+
+        # build the network which we can submit
+        fec_network = planned_network.to_alchemical_network()
+        # build the network on alchemiscale and get a key
+        network_key = self._client.create_network(fec_network, scope)
+
+        # save the scoped key to find the network later and build the planned network again
+        alchem_result = AlchemiscaleResults(network_key=network_key)
+        network_data["results"] = alchem_result
+        # work around the frozen object
+        result_network = FreeEnergyCalculationNetwork(**network_data)
+        return result_network
+
+    def action_network(
+        self, planned_network: FreeEnergyCalculationNetwork
+    ) -> list[Optional[ScopedKey]]:
+        """
+        For the given network which is already stored on alchemiscale create and action tasks.
+
+        Args:
+            planned_network: The network which should action tasks for.
+
+        Returns:
+            A list of actioned tasks for this network.
+
+        """
+        network_key = planned_network.results.network_key
+
+        tasks = []
+        for tf_sk in self._client.get_network_transformations(network_key):
+            # for each task create x repeats which are configured by the factory
+            tasks.extend(
+                self._client.create_tasks(tf_sk, count=planned_network.n_repeats + 1)
```

Comment:
Why is the +1 needed here?

Reply:
This is due to the factory recording how many times each transform should be repeated, so the total number of tasks is 1 + `n_repeats` we can change this field to be total runs if that would be more clear.

Reply:
happy either way